pkg/controllers/node/termination/suite_test.go

```diff
@@ -707,10 +709,65 @@ var _ = Describe("Termination", func() {
 				g.Expect(pod.DeletionTimestamp.IsZero()).To(BeTrue())
 			}, ReconcilerPropagationTime, RequestInterval).Should(Succeed())
 		})
+		It("should preemptively delete pods to satisfy their terminationGracePeriodSeconds", func() {
+			nodeClaim.Spec.TerminationGracePeriod = &metav1.Duration{Duration: time.Second * 300}
+			node.Annotations = map[string]string{
+				v1.NodeTerminationTimestampAnnotationKey: time.Now().Add(nodeClaim.Spec.TerminationGracePeriod.Duration).Format(time.RFC3339),
+			}
+			pod := test.Pod(test.PodOptions{
+				NodeName: node.Name,
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1.DoNotDisruptAnnotationKey: "true",
+					},
+					OwnerReferences: defaultOwnerRefs,
+				},
+				TerminationGracePeriodSeconds: lo.ToPtr(int64(300)),
+			})
+			ExpectApplied(ctx, env.Client, node, nodeClaim, nodePool, pod)
+
+			Expect(env.Client.Delete(ctx, node)).To(Succeed())
+			ExpectObjectReconciled(ctx, env.Client, terminationController, node)
+			ExpectSingletonReconciled(ctx, queue)
+			ExpectNodeExists(ctx, env.Client, node.Name)
+			ExpectDeleted(ctx, env.Client, pod)
+		})
+		It("should not delete pods if their terminationGracePeriodSeconds is less than the the node's remaining terminationGracePeriod", func() {
```

Comment:
nit on naming since it tests both
```suggestion
		It("should only delete pods when their terminationGracePeriodSeconds is less than the the node's remaining terminationGracePeriod", func() {
```